### DISCUSSION TITLE
Axes widget bugs in local view

### DISCUSSION THREAD

**Discussion by karelvrabeckv** [Kitware/trame#466] [Trame/VTK] [answered]
Hello,
Is it possible to show the labels of `vtkAxesActor` in `VtkLocalView` mode?
Currently, there are no labels even if I use `AxisLabelsOn()` or `SetNAxisLabelText(...)`.

![widget](https://github.com/Kitware/trame/assets/62759688/bb65358d-148b-43e6-92d9-89e6e14f9b65)

Additionally, this widget suddenly disappears if I start zooming with the rotation wheel of my mouse. But it only happens in the beginning if I haven't interacted yet. Is this also a bug of `vtkAxesActor` or am I doing something wrong?

```
def create_axes_widget(self, render_window_interactor: vtkRenderWindowInteractor):
    axes_widget = vtkOrientationMarkerWidget()
        
    axes_widget.SetOrientationMarker(vtkAxesActor())
    axes_widget.SetInteractor(render_window_interactor)
    axes_widget.SetViewport(0.0, 0.0, 0.35, 0.35)
        
    axes_widget.EnabledOn()
        
    return axes_widget
```

Thank you very much! :)

**Comment by jourdain** [comment, marked as answer]
Just a limitation of the vtk.js implementation. Should be fixed soon with the usage of VTK/wasm for a VtkLocalView implementation.

**Reply by karelvrabeckv** [reply]
Hello, is there already some progress? Thank you.

**Reply by jourdain** [reply]
Waiting on some CI to complete for registry publication of the WASM piece. Once that's done, we can update and release the trame module for it.

**Reply by FBIShowYourHand** [reply]
Now it still doesn't seem to work.

**Reply by jourdain** [reply]
Status: 
- The orientation widget works with vtk.js but lack labels
- The orientation widget with labels works with VTK.wasm

The picture below show the rendering done with wasm 
![Screenshot 2025-06-19 at 08 49 36](https://github.com/user-attachments/assets/470f893c-f877-4be9-977f-19eddfaf422e)
